### PR TITLE
verify reservation

### PR DESCRIPTION
**Pull request by PalmiraBenavente**
- added reservation verification

- verify that the data is not null

- verify that reservations do not overlap

- addition of enumerable to identify each error

- verification abstraction in a class

image
image

https://user-images.githubusercontent.com/84336438/124970915-f3ce4180-dffe-11eb-9a1c-2f47df3e566e.mp4

https://user-images.githubusercontent.com/84336438/124971713-ec5b6800-dfff-11eb-903f-2c59409eb3b6.mp4

https://user-images.githubusercontent.com/84336438/124971919-26c50500-e000-11eb-9858-9220ab7ef261.mp4

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystem/activities/ParkingReservationActivity.java [hunk 1]
```diff
@@ -23,7 +24,7 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         binding = ActivityReserverParkingLotBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
-        presenter = new ReservationParkingPresenter(new ReservationParkingView(this, binding), new ReservationParkingModel(ParkingDatabase.getInstance()));
+        presenter = new ReservationParkingPresenter(new ReservationParkingView(this, binding), new ReservationParkingModel(ParkingDatabase.getInstance(), new ReservationChecker(ParkingDatabase.getInstance())));
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystem/activities/ParkingReservationActivity.java:27]: nice!

File: app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    private boolean reservationOK(Reservation reservation) {
+        int lot = reservation.getParkingLot();
+        if (database.getHashMapReservation().containsKey(lot)) {
+            List<Reservation> reservationsOnLot = database.getHashMapReservation().get(lot);
+            for (Reservation i : reservationsOnLot) {
+                if (isOverlap(i, reservation)) {
+                    return false;
+                }
+            }
+            return true;
```
Comment (lucianoeli) [inline app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java]: you could remove else block and just return true outside if block

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java (hunk 2 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java]: Done

File: app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java [hunk 3]
```diff
@@ -43,6 +43,38 @@ public void toastShowTextData(int lote, int code, String dateStart, String dateE
         Activity activity = getActivity();
         if (context != null && activity != null)
             Toast.makeText(context, context.getString(R.string.toast_resaervation_parking, lote, code, dateStart, dateEnd), Toast.LENGTH_SHORT).show();
-            activity.onBackPressed();
+        activity.onBackPressed();
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java]: missing brackets, you're not checking if activity is null

File: app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        if (reservation.getStartDate() == null) {
+            return EnumReservationVerify.MISSING_DATESTART;
+        }
+        if (reservation.getEndDate() == null) {
+            return EnumReservationVerify.MISSING_DATEEND;
+        }
+        if (reservation.getParkingLot() == Constants.INT_MINUS_ONE) {
+            return EnumReservationVerify.MISSING_LOT;
+        }
+        if (reservation.getUserCode() == Constants.INT_MINUS_ONE) {
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java]: please use descriptive names for constants

File: app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java (hunk 3 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/mvp/view/ReservationParkingView.java]: Done

File: app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java (hunk 4 above)
Comment (PalmiraBenavente) [reply app/src/main/java/com/example/parkingsystem/utils/ReservationChecker.java]: Done

**Review by nicolbidone — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**